src/main/java/org/hydev/mcpm/client/database/SuperLocalPluginTracker.java

```diff
@@ -0,0 +1,619 @@
+
+        HashMap<String, PluginTrackerModel> currentListName = new HashMap<>();
+
+        for (PluginTrackerModel pluginTrackerModel : currentList) {
+            currentListName.put(pluginTrackerModel.name, pluginTrackerModel);
+        }
+
+        List<PluginYml> installedPlugins = listInstalled();
+
+        // Iterate through each plugin in installedPlugins. If the plugin exists in both
+        // currentList and installedPlugins
+        // add its representation in currentList to toAdd. If the plugin exists only in
+        // installedPlugins, instantiate
+        // a new PluginTrackerModel representing this newly-installed plugin, with
+        // default manuallyInstalled false.
+
+        ArrayList<PluginTrackerModel> toAdd = new ArrayList<>();
+
+        for (PluginYml pluginYml : installedPlugins) {
+            PluginTrackerModel PluginRepresentation = new PluginTrackerModel(pluginYml.name(), false,
+                    pluginYml.version(), "unknown");
+
+            // if the plugin exists in currentList and installedPlugins,
+            // add it to toAdd with its currently stored parameters and status
+
+            if (currentListName.containsKey(PluginRepresentation.name)) {
+                if (currentListName.get(PluginRepresentation.name).versionId.equals(PluginRepresentation.versionId)) {
+                    // if the plugin exists in both currentList and installedPlugins, pass it on
+                    toAdd.add(currentListName.get(PluginRepresentation.name));
+                } else if (tryPreserveLocalStatus) {
+                    // if the plugin exists in both currentList and installedPlugins, but the
+                    // version differs
+                    // and we are trying to preserve local status, pass it on with the same status
+                    PluginRepresentation.isManual = currentListName.get(PluginRepresentation.name).isManual;
+                }
+                // Overwrite the local representation with a new one
+                toAdd.add(PluginRepresentation);
+
+            } else {
+                toAdd.add(PluginRepresentation);
+            }
+        }
+
+        saveCsv(toAdd);
+    }
+
+    /**
+     * List all currently installed plugins in an ArrayList
+     */
+    public List<PluginYml> listInstalled() {
+        // Go into the plugin directory and list files
+        File dir = new File(pluginDirectory);
+        File[] list = dir.listFiles();
+        if (list == null)
+            return new ArrayList<>();
+
+        // Filter only java files, return all metadata that's not null
+        return Arrays.stream(list).filter(f -> f.isFile() && f.getName().endsWith(".jar"))
+                .map(this::readMeta).filter(Objects::nonNull).toList();
+    }
+
+    /**
+     * Mark a plugin as manually installed (as opposed to a dependency)
+     * Precondition: mainLockFile is a sorted .csv file with the following format:
```

Comment:
sorted is a precondition but I don't see the precondition being used or enforced anywhere. Maybe you can just remove it